src/system_database.ts

```diff
@@ -385,146 +608,78 @@ export class PostgresSystemDatabase implements SystemDatabase {
-        request,
-        executor_id,
-        application_version,
-        application_id,
-        created_at,
-        recovery_attempts,
-        updated_at
-      ) VALUES($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14, $15, $16)
-       ON CONFLICT (workflow_uuid)
-        DO UPDATE SET
-          recovery_attempts = workflow_status.recovery_attempts + 1,
-          updated_at = EXCLUDED.updated_at,
-          executor_id = EXCLUDED.executor_id 
-        RETURNING recovery_attempts, status, name, class_name, config_name, queue_name`,
-      [
-        initStatus.workflowUUID,
-        initStatus.status,
-        initStatus.workflowName,
-        initStatus.workflowClassName,
-        initStatus.workflowConfigName,
-        initStatus.queueName,
-        initStatus.authenticatedUser,
-        initStatus.assumedRole,
-        JSON.stringify(initStatus.authenticatedRoles),
-        JSON.stringify(initStatus.request),
-        initStatus.executorId,
-        initStatus.applicationVersion,
-        initStatus.applicationID,
-        initStatus.createdAt,
-        initStatus.status === StatusString.ENQUEUED ? 0 : 1,
-        Date.now(),
-      ],
-    );
-    // Check the started workflow matches the expected name, class_name, config_name, and queue_name
-    // A mismatch indicates a workflow starting with the same UUID but different functions, which should not be allowed.
-    const resRow = result.rows[0];
-    initStatus.workflowConfigName = initStatus.workflowConfigName || '';
-    resRow.config_name = resRow.config_name || '';
-    resRow.queue_name = resRow.queue_name === null ? undefined : resRow.queue_name; // Convert null in SQL to undefined
-    let msg = '';
-    if (resRow.name !== initStatus.workflowName) {
-      msg = `Workflow already exists with a different function name: ${resRow.name}, but the provided function name is: ${initStatus.workflowName}`;
-    } else if (resRow.class_name !== initStatus.workflowClassName) {
-      msg = `Workflow already exists with a different class name: ${resRow.class_name}, but the provided class name is: ${initStatus.workflowClassName}`;
-    } else if (resRow.config_name !== initStatus.workflowConfigName) {
-      msg = `Workflow already exists with a different class configuration: ${resRow.config_name}, but the provided class configuration is: ${initStatus.workflowConfigName}`;
-    } else if (resRow.queue_name !== initStatus.queueName) {
-      // This is a warning because a different queue name is not necessarily an error.
-      this.logger.warn(
-        `Workflow (${initStatus.workflowUUID}) already exists in queue: ${resRow.queue_name}, but the provided queue name is: ${initStatus.queueName}. The queue is not updated. ${new Error().stack}`,
-      );
-    }
-    if (msg !== '') {
-      throw new DBOSConflictingWorkflowError(initStatus.workflowUUID, msg);
-    }
+    const client = await this.pool.connect();
+    try {
+      await client.query('BEGIN');
```

Comment:
Isolation Level?

Reply:
Default read committed is fine. We need repeatable read level for transactions that first read and then write. So only `resumeWorkflow` and `findAndMarkStartableWorkflows` need repeatable read -- which are already marked so.